install.sh

```diff
@@ -1,34 +1,49 @@
 #!/bin/sh
 set -e
-if [ "$1" != "" ]; then
-	if [ -d "$1" ]; then
-		mkdir -p $1/tools/agbcc
-		mkdir -p $1/tools/agbcc/bin
-		mkdir -p $1/tools/agbcc/include
-		mkdir -p $1/tools/agbcc/lib
-		cp agbcc $1/tools/agbcc/bin/
-		cp old_agbcc $1/tools/agbcc/bin/
-		cp agbcc_arm $1/tools/agbcc/bin/
-		cp -R libc/include $1/tools/agbcc/ #drop include, because we don't want include/include
-		cp ginclude/* $1/tools/agbcc/include/
-		cp libgcc.a $1/tools/agbcc/lib/
-		cp libc.a $1/tools/agbcc/lib/
-		echo "agbcc successfully installed!"
+
+if [ "$1" = "-h" ]; then
+	echo "Usage: install.sh PATH"
+	exit 0
+fi
+
+# If there's no installation directory specified, the install script will default to 
+# installing agbcc into /opt/pret/agbcc
+if [ "$1" = "" ]; then
+	BASE_INSTALL_DIRECTORY=/opt/pret
+	INSTALL_DIRECTORY=$BASE_INSTALL_DIRECTORY
+	mkdir -p $INSTALL_DIRECTORY
+else
+	BASE_INSTALL_DIRECTORY=$1
+	INSTALL_DIRECTORY=$BASE_INSTALL_DIRECTORY/tools
+fi
+
+# The BASE_INSTALL_DIRECTORY nomenclature is so that the check for the existence of the 
+# directory doesn't fail just because there's no tools directory yet.
+if [ -d "$BASE_INSTALL_DIRECTORY" ]; then
+	mkdir -p $INSTALL_DIRECTORY/agbcc
+	mkdir -p $INSTALL_DIRECTORY/agbcc/bin
+	mkdir -p $INSTALL_DIRECTORY/agbcc/include
+	mkdir -p $INSTALL_DIRECTORY/agbcc/lib
+	cp agbcc $INSTALL_DIRECTORY/agbcc/bin/
+	cp old_agbcc $INSTALL_DIRECTORY/agbcc/bin/
+	cp agbcc_arm $INSTALL_DIRECTORY/agbcc/bin/
+	cp -R libc/include $INSTALL_DIRECTORY/agbcc/ #drop include, because we don't want include/include
+	cp ginclude/* $INSTALL_DIRECTORY/agbcc/include/
+	cp libgcc.a $INSTALL_DIRECTORY/agbcc/lib/
+	cp libc.a $INSTALL_DIRECTORY/agbcc/lib/
+	echo "agbcc successfully installed!"
```

Comment:
```suggestion
	install -v -Dm755 -t "$INSTALL_DIRECTORY/bin" agbcc old_agbcc agbcc_arm
	install -v -Dm644 -t "$INSTALL_DIRECTORY/lib" libgcc.a libc.a
	install -v -Dm644 -t "$INSTALL_DIRECTORY/include" ginclude/*
	cp -vR libc/include "$INSTALL_DIRECTORY/" #drop include, because we don't want include/include
	echo "agbcc successfully installed!"
```
More verbosity, and explicit file permissions